libchihua-msg/src/node.c

```diff
@@ -0,0 +1,81 @@
+#include "../include/node.h"
+#include "../include/node/peer-dbus.h"
+#include "./common/memory.h"
+
+#include <stdio.h>
+
+Node *node_new(const NodeParams *p) {
+        fprintf(stdout, "Creating Node...\n");
+
+        int r = 0;
+        sd_event *event = NULL;
+        r = sd_event_new(&event);
+        if (r < 0) {
+                fprintf(stderr, "Failed to create event loop: %s\n", strerror(-r));
+                return NULL;
+        }
+
+        char *orch_addr = assemble_address(p->orch_addr);
+        if (orch_addr == NULL) {
+                return NULL;
+        }
+
+        Node *n = malloc0(sizeof(Node));
+        n->event_loop = steal_pointer(&event);
+        n->orch_addr = orch_addr;
+
+        return n;
+}
+
+void node_unrefp(Node **n) {
```

Comment:
To keep the style consistent with the POC, consider having this function take a pointer to Node, and then use ```_SD_DEFINE_POINTER_CLEANUP_FUNC``` to define a cleanup function that takes a pointer to pointer to Node.

Reply:
Oh, I missed `_SD_DEFINE_POINTER_CLEANUP_FUNC`. I'll check it out, thanks!